pkg/apis/kops/instancegroup.go

```diff
@@ -82,8 +82,17 @@ const (
 // SupportedFilesystems is a list of supported filesystems to format as
 var SupportedFilesystems = []string{BtfsFilesystem, Ext4Filesystem, XFSFilesystem}
 
+type InstanceManager string
+
+const (
+	InstanceManagerCloudGroup InstanceManager = "CloudGroup"
+	InstanceManagerKarpenter  InstanceManager = "Karpenter"
+)
+
 // InstanceGroupSpec is the specification for an InstanceGroup
 type InstanceGroupSpec struct {
+	// InstanceManager determines what is managing the node lifecycle
+	InstanceManager InstanceManager `json:"instanceManager,omitempty"`
```

Comment:
I think `provisioner` is a better name. If not that, the "instance" part is redundant, suggesting `manager`.

Reply:
Sounds good. I think I will go with `manager`